### ISSUE TITLE
Dropdown Refactor

### ISSUE THREAD

**Issue by frattaro** [cisco-sbg-ui/atomic-react#40]
**Describe the bug**

The `ADropdown` component isn't keyboard accessible.

**To Reproduce**

Bring the focus onto an `ADropdown` menu, press the `esc` key. The menu doesn't close.

**Expected behavior**

The dropdown opener should:
- on `Up Arrow`, open the menu and focus the last menu item
- on `[Down Arrow, Space, Enter]`, open the menu and focus the first menu item

Each dropdown menu item should:
- on `Enter`, activate the menu item, then close the menu and set focus to the opener.
- on `Escape`, close the menu and set focus to the opener.
- on `Up Arrow`, move focus to the previous menu item. If the focus is on the first menu item, move focus to the last menu item.
- on `Down Arrow`, move focus to the next menu item. If the focus is on the last menu item, move focus to the first menu item.

**Additional context**

These changes significantly alter the `ADropdown` interface. The dropdown needs to know what element is the opener and it does not, currently.

If the dropdown knows what the opener element is, it should be able to apply the following attributes: `aria-haspopup`, `aria-expanded`, `aria-controls`, `role="button" tabIndex={0}` (if not a button). It should also be able to add event listeners automatically (onClick, onKeyDown).

The behavior would vary if the menu has non-interactable children (they may all lack a `href` or `onClick` handler).
